File: drafts/bs-Zarith/lib/js/__tests__/bs_Z_Int64_test.js
'use strict';

var Z = require("../src/Z.js");
var Jest = require("@glennsl/bs-jest/lib/js/src/jest.js");

Jest.describe("Z.Int64", (function (param) {
        Jest.test("add", (function (param) {
                var x = Z.Int64.of_string("230000232323322323");
                var y = Z.Int64.of_int(3);
                var z = Z.Int64.sub(x, y);
                return Jest.Expect.toEqual(Z.Int64.of_string("230000232323322320"), Jest.Expect.expect(z));
              }));
      }));

/*  Not a pure module */

File: drafts/bs-Zarith/__tests__/bs_Z_Int64_test.ml
open Jest
open Expect

let () = 

describe "Z.Int64" (fun () ->
    test "add" (fun () ->
        let x = Z.Int64.of_string "230000232323322323" in
        let y = Z.Int64.of_int 3 in
        let z = Z.Int64.(sub x y) in

        expect @@ z |> toEqual (Z.Int64.of_string "230000232323322320"));

  )
